packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -3015,6 +3032,13 @@ class TextSelectionGestureDetector extends StatefulWidget {
     required this.child,
   });
 
+  /// Called whenever [TapAndDragGestureRecognizer] has started tracking
+  /// a tap gesture.
+  final VoidCallback? onTapTrackStart;
+
+  /// Called when [TapAndDragGestureRecognizer] tap tracking has been reset.
```

Comment:
Nit: maybe we should reference `BaseTapAndDragGestureRecognizer` here instead.

Reply:
We should also point out when this happens. For example of the next pointer down event, when the duration of [kDoubleTapTimeout] has elapsed between a [PointerUpEvent] and the following [PointerDownEvent].

Reply:
I see you already do this in the gestures documentation. Maybe we can just make a template and insert it here? Same for `onTapTrackStart`.

Reply:
Ah yes, I prefer macros too.